generation/hermetic_split_repo.sh

```diff
@@ -0,0 +1,209 @@
+    echo "Could not generate owl-bot-staging via copy-bazel-bin."
+    exit 1
+  fi
+
+  # Extracts files from staging directory and post-process
+  # the files via OwlBot. This includes templating, formatting, etc.
+  # The latest owlbot-java fails with processing old repository that did not have
+  # api_shortname field in .repo-metadata.json. The sha256 is the oldest one I found in gcr.io.
+  # This version did not reqiure api_shortname field in .repo-metadata.json.
+  OWLBOT_LOCK_FILE=workspace/repo/.github/.OwlBot.lock.yaml
+  if [ -r "$OWLBOT_LOCK_FILE" ]; then
+    OWLBOT_JAVA_VERSION=$(grep 'digest:' "${OWLBOT_LOCK_FILE}"  |awk '{ print $2 }')
+    echo "Found ${OWLBOT_LOCK_FILE} with ${OWLBOT_JAVA_VERSION} as owlbot-java version"
+  else
+    echo "No ${OWLBOT_LOCK_FILE}"
+  fi
+  docker run --rm --user "$(id -u):$(id -g)" \
+      -v "$(pwd)/workspace/repo:/workspace"   -w /workspace \
+      "gcr.io/cloud-devrel-public-resources/owlbot-java@${OWLBOT_JAVA_VERSION}"
+}
+
+function regenerate_with_synthtool() {
+  cp hermetic_synthtool_command.sh workspace/
+
+  echo "Preparing an old synthtool revision"
+
+  git clone https://github.com/googleapis/synthtool workspace/synthtool
+
+  pushd workspace/synthtool
+
+  # Sort the commits by the date. Choose the last one.
+  last_synthtool_commit=$(for C in $synthtool_commits; do
+    git --no-pager log -1 --format='%H %at %ai' $C
+  done |sort -k 2 -r |head -1 |cut -d ' ' -f1)
+  echo "The last synthtool commit on the branch ${BRANCH} was:"
+  git log --pretty=format:"%h%x09%an%x09%ad%x09%s" -1 $last_synthtool_commit
+  echo "Checking out this commit from googleapis/synthtool repository:"
+  last_synthtool_commit_short=${last_synthtool_commit:0:7}
+  git checkout -B "branch-from-${last_synthtool_commit_short}" ${last_synthtool_commit_short}
+  popd
+
+  docker run --rm  --user "$(id -u):$(id -g)" --env HOME=/workspace \
+      --env USER=$(id -u -n) -v $(pwd)/workspace:/workspace \
+      -w /workspace/repo \
+      --entrypoint "/workspace/hermetic_synthtool_command.sh" \
+      -it gcr.io/repo-automation-bots/bazel-bot@sha256:bfdcd03c8ce2aec6e3663161a2c9ef5356594780c506011d35d62eecbd3b447e
+}
+
+if [ "true" = "${use_synthtool}" ]; then
+  regenerate_with_synthtool
+else
+  regenerate_with_owlbot
+fi
+
+# Exclude irrelevant changes
+pushd workspace/repo
+git checkout "origin/${BRANCH}" -- README.md .github .kokoro java.header renovate.json \
+    versions.txt ./**/*.xml
+# java-commons-protos doesn't have samples
+if [ -d samples ]; then
+  git checkout "origin/${BRANCH}" -- samples
+fi
+
+git add ./**/*.java
+
+# It's possible that there's no code change.
+git commit -m 'fix: applying a different code generation' --allow-empty
+popd
+
+echo "Changes are ready in workspace/repo. Create a pull request: "
+echo "  cd workspace/repo && gh pr create --draft --base ${BRANCH} \
+    --title 'deps: test building with newer library version'" \
+    --body "'No need to review.'"
```

Comment:
minor comment: I had to remove the `echo` part when running this command locally to create a draft PR. However, it may behave differently within the script.

Reply:
As discussed offline: this line is only meant to contain instructions on how to create a draft PR once the `hermetic_split_repo` script finishes running.